cmd/argo-watcher/argo_status_updater.go

```diff
@@ -76,6 +78,20 @@ func (updater *ArgoStatusUpdater) waitForApplicationDeployment(task models.Task)
 	var application *models.Application
 	var err error
 
+	app, err := updater.argo.api.GetApplication(task.App)
+	if err != nil {
+		return nil, err
+	}
+
+	if app.IsManagedByWatcher() && task.Validated {
+		updater.Lock()
+		defer updater.Unlock()
+		log.Debug().Str("id", task.Id).Msg("Application managed by watcher. Initiating git repo update.")
+		app.UpdateGitImageTag(&task)
+	} else {
+		log.Debug().Str("id", task.Id).Msg("Skipping git repo update: Application not managed by watcher or token is absent/invalid.")
```

Comment:
Again, I'd fail the API call when token is invalid. Then this message would only be for tasks that were not requested to be deployed with git commit